packages/7-todo-tdd/src/components/todo-app/TodoApp.test.tsx

```diff
@@ -0,0 +1,157 @@
+import { ReactWrapper } from 'enzyme';
+import { wrapperGenerator } from '../../styles/utils';
+import { TodoAppDriver } from './TodoApp.test.driver';
+import { s } from './TodoApp';
+import { Chance } from 'chance';
+import { TodoItemDriver } from '../todo-item/TodoItem.test.driver';
+import { todosService } from '../../services/TodoService';
+import { ITodoDTO } from 'fed-todo-journey_todo-common';
+
+const chance = new Chance();
+const h = wrapperGenerator('.');
+
+
+describe('app main components should be rendered', () => {
+    let mountedTodoApp: ReactWrapper;
+
+    beforeEach(async () => {
+        const todosDriver: TodoAppDriver = new TodoAppDriver();
+        await todosDriver.waitForAppToUpdate();
+        mountedTodoApp = todosDriver.getAppComponent();
+    });
+
+    it('app should be rendered', () => {
+        expect(mountedTodoApp.exists()).toBe(true);
+    });
+
+    it('todo input should be rendered', () => {
+        expect(mountedTodoApp.find(h(s.todo__inputContainer__input)).exists()).toBe(true);
+    });
+
+    it('todo list should be rendered', () => {
+        expect(mountedTodoApp.find(h(s.todo__list)).exists()).toBe(true);
+    });
+
+    it('add btn should be rendered', () => {
+        expect(mountedTodoApp.find(h(s.todo__inputContainer__addBtn)).exists()).toBe(true);
+    });
+});
+
+describe('checks todo list - add functionality', () => {
+    let appTestDriver: TodoAppDriver;
+
+    beforeEach(() => {
+        appTestDriver = new TodoAppDriver();
+    });
+
+    afterEach(async ()=>{
+        appTestDriver.removeTodosFromFakeTodosService();
+    })
+
+    it('should not add todo item with empty content', async () => {
+        appTestDriver.todoInputInsertContent('');
+        appTestDriver.clickOnAddBtn();
+        await appTestDriver.waitForAppToUpdate();
+        expect(appTestDriver.getTodosCount()).toBe(0);
+    });
+
+    it('should add todo item with content when pressing enter', async () => {
+        appTestDriver.todoInputInsertContent('example');
+        appTestDriver.pressEnterOnTodoInput();
+        await appTestDriver.waitForAppToUpdate()
+        expect(appTestDriver.getTodosCount()).toBe(1);
```

Comment:
a better check would be checking that the actual todo item is the one that we entered.. either by checking the content "example" or better yet use some random content to be sure.

again, missing the given part. you are relaying on the fact that when the test starts the todos are empty.